app/controllers/UploadSchemaController.java

```diff
@@ -51,4 +53,18 @@ public Result getUploadSchema(String schemaId) {
         UploadSchema uploadSchema = uploadSchemaService.getUploadSchema(studyId, schemaId);
         return okResult(uploadSchema);
     }
+
+    /**
+     * Play controller for GET /researcher/v1/uploadSchema/forStudy. This API fetches all revisions of all upload
+     * schemas in the current study. This is generally used by worker apps to validate uploads against schemas.
+     * @return Play result with list of schemas for this study
+     */
+    public Result getUploadSchemasForStudy() throws Exception {
+        // TODO: When we implement worker accounts, they should have access to the API as well.
+        UserSession session = getAuthenticatedResearcherOrAdminSession();
+        String studyId = session.getStudyIdentifier().getIdentifier();
+
+        List<UploadSchema> schemasByIdAndRev = uploadSchemaService.getUploadSchemasForStudy(studyId);
```

Comment:
If this takes a study ID I think it should take StudyIdentifier for type safety. Otherwise everything in our system takes a string ID of this that or whatever.